lektor/admin/static/js/components/Component.jsx

```diff
@@ -1,17 +1,17 @@
 'use strict'
 import dialogSystem from '../dialogSystem'
 import BaseComponent from './BaseComponent'
+import PropTypes from 'prop-types'
 
 class Component extends BaseComponent {
-  constructor (props) {
-    super(props)
+  constructor (props, context) {
+    super(props, context)
```

Comment:
I'm not really sure how context works here. This should call `BaseComponent`'s constructor here (which is the same as `React.Component`). I can't find anything about a second argument for it (https://reactjs.org/docs/react-component.html#constructor). Where is this passed from?

Reply:
See https://reactjs.org/docs/legacy-context.html#referencing-context-in-lifecycle-methods - I didn't realise it, but it's actually a legacy API that will be removed after React 16. I don't think that's a huge issue though, as it works, and as mentioned this is a stopgap fix to get react to 16, rather than a rewrite to make it work with 16 concepts and APIs.

Reply:
I agree, let's get this merged.